dev/flipbook.tsx

```diff
@@ -64,6 +82,15 @@ export function Flipbook() {
     const questionsState = state.questions.trim();
     const noTextEntered = questionsState === "";
 
+    const graphieImageUrls = useMemo<ReadonlyArray<string>>(
+        () =>
+            Object.values(question?.widgets ?? {})
+                .filter(interactiveGraphPredicate)
+                .map((w) => w.options.backgroundImage?.url ?? "")
+                .filter((url) => url.length > 0),
```

Comment:
Empty string, null, and undefined are all falsey, so this could be simplified to
```suggestion
                .map((w) => w.options.backgroundImage?.url)
                .filter(Boolean),
```

That might be too much magic, though?

Reply:
It'd love to do that, but Typescript doesn't seem to understand the narrowing that occurs in `.filter(Boolean)`. 

https://www.typescriptlang.org/play/?#code/C4TwDgpgBA8mwEsD2A7AzlAvFA3gKCigCMBDAYwGsBzAJyQFcUATASQFsSqIB+ALigBC5anUasOXANx4AvnjyhIg4bQbN2naNnyF6NADb80wGghRUoAHygp6+-VahiIAMzMQms+YugtgENixcAigkeGR0fjhEVDQvPDJY4CgEfzY0fgAlCBImVH0QAEEaGhIQAB4-AIA+IIBtAF15PFSAtAA6DjAACgQsWoR2sJj0dtJKVTENLm52vX0ASna3fX8aboEkJH0clAWoNBJENDcIDGzc-KKSsvLjU3Nq6SA